features/stop_on_failure_configuration.feature

```diff
@@ -0,0 +1,113 @@
+Feature: Stop on failure via configuration
+  In order to stop further execution of steps when first step fails
+  As a feature developer
+  I need to have a configuration.stop_on_failure sets to true
+
+  Background:
+    Given a file named "features/bootstrap/FeatureContext.php" with:
+      """
+      <?php
+
+      use Behat\Behat\Context\Context;
+
+      class FeatureContext implements Context
+      {
+          /**
+           * @Given /^I have (?:a|another) step that passes?$/
+           * @Then /^I should have a scenario that passed$/
+           */
+          public function passing() {
+          }
+
+          /**
+           * @Given /^I have (?:a|another) step that fails?$/
+           * @Then /^I should have a scenario that failed$/
+           */
+          public function failing() {
+              throw new Exception("step failed as supposed");
+          }
+
+      }
+      """
+    And a file named "features/failing.feature" with:
+      """
+      Feature: Failing Feature
+        In order to test the stop-on-failure feature
+        As a behat developer
+        I need to have a feature that fails
+
+        Background:
+          Given I have a step that passes
+
+        Scenario: 1st Passing
+          When I have a step that passes
+          Then I should have a scenario that passed
+
+        Scenario: 1st Failing
+          When I have a step that passes
+           And I have another step that fails
+          Then I should have a scenario that failed
+
+        Scenario: 2st Failing
+          When I have a step that fails
+          Then I should have a scenario that failed
+
+        Scenario: 2nd Passing
+          When I have a step that passes
+           And I have another step that passes
+          Then I should have a scenario that passed
+      """
+
+  Scenario: with stop_on_failure set to false
+   Given a file named "behat.yml" with:
+      """
+      default:
+        configuration:
+          stop_on_failure: false
```

Comment:
Oh also, although we generally use `_` for map keys, perhaps we should use `-` (e.g. `stop-on-failure`) for these so that they exactly match the CLI arguments?

Reply:
I questioned myself on that, and preferred to be consistent with the rest of the yaml configuration. But it joins the other discussion about the naming, because it's the same philosophy.

Reply:
I think I would find it weird to see `-` used here when we use `_` everywhere else

Reply:
The relationship between these options and the command line ones should be obvious